File: dataset/human/c/file_1760.c
#include <math.h>
#include <stdlib.h>
#include <string.h>
#include <stdio.h>

int main()
{   
    int n;
    scanf("%d",&n);
    int a[50],b[50],c[50];
    for(int i=0;i<n;i++){
        scanf("%d",&a[i]);
    }
    for(int i=0;i<n;i++){
        scanf("%d",&b[i]);
    }
    for(int i=0;i<n-1;i++){
        scanf("%d",&c[i]);
    }
    int ans=0;
    for(int i=0;i<n;i++){
        ans+=b[a[i]-1];
    }
    for(int i=0;i<n-1;i++){
        if(a[i+1]==a[i]+1){
            ans+=c[a[i]-1];
        }
    }
   
    printf("%d",ans);

    return 0;
}

File: dataset/human/python/file_1760.py
import sys
import copy
import string
import  math
from _bisect import *
from collections import *
from operator import itemgetter
from math import factorial
"""
from fractions import gcd
def lcm(x, y):
    return (x * y) // gcd(x, y)
"""
stdin = sys.stdin
ni = lambda: int(ns())
na = lambda: list(map(int, stdin.readline().split()))
ns = lambda: stdin.readline()

n, m = na()
a = [list(input()) for _ in range(n)]
b = [list(input()) for _ in range(m)]
ans = 'No'
for i in range(n-m+1):
    for j in range(n-m+1):
        flag = True
        for k in range(m):
            for l in range(m):
                if a[i + k][j + l] != b[k][l]:
                    flag = False
                    break
        if flag:
            ans = 'Yes'
print(ans)
